CMakeLists.txt

```diff
@@ -122,22 +155,70 @@ if(CCPP)
+set_property(SOURCE FMS/fft/fft.F90 APPEND_STRING PROPERTY COMPILE_FLAGS "-DSGICRAY=0 -DNAGFFT=0")
+
+target_include_directories(fms PUBLIC $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/FMS/include>
+                                      $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/FMS/fms>
+                                      $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/FMS/fms2_io/include>
+                                      $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/FMS/mpp/include>)
+
+target_link_libraries(fms PRIVATE NetCDF::NetCDF_Fortran)
+set_target_properties(fms PROPERTIES ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/FMS)
+set_target_properties(fms PROPERTIES Fortran_MODULE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/FMS/mod)
+target_include_directories(fms PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/FMS/mod>)
+
+###############################################################################
+### CMEPS
+###############################################################################
+if(CMEPS)
+    include(cmeps_files.cmake)
+    add_library(cmeps ${cmeps_src_files})
+
+    target_include_directories(cmeps PRIVATE $ENV{PIO_INCLUDES})
+    target_include_directories(cmeps PUBLIC $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/CMEPS/mediator>)
+
+    target_link_libraries(cmeps PUBLIC $ENV{PIO_LIBRARIES}/libpiof.a)
+    target_link_libraries(cmeps PUBLIC $ENV{PIO_LIBRARIES}/libpioc.a)
+    if(ESMF_FOUND)
+        target_link_libraries(cmeps PUBLIC esmf)
+    endif()
+
+    target_compile_definitions(cmeps PRIVATE "-DINTERNAL_PIO_INIT")
+    set_target_properties(cmeps PROPERTIES ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/CMEPS)
+    set_target_properties(cmeps PROPERTIES Fortran_MODULE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/CMEPS/mod)
+    target_include_directories(cmeps PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/CMEPS/mod>)
+endif()
+###############################################################################
+### stochastic_physics
+###############################################################################
+add_subdirectory(stochastic_physics)
 
 ###############################################################################
 ### FV3
 ###############################################################################
 add_subdirectory(FV3)
 
 ###############################################################################
-### MOM6
+### CICE_interface
 ###############################################################################
-add_subdirectory(MOM6-interface)
+if(CICE6)
+    add_subdirectory(CICE-interface)
+endif()
 
 ###############################################################################
-### CICE
+### MOM6-interface
 ###############################################################################
-add_subdirectory(CICE-interface)
+if(MOM6)
+    add_subdirectory(MOM6-interface)
+endif()
 
 ###############################################################################
 ### WW3
```

Comment:
and possibly WW3 too.
In the S2S model all these components are not optional.

Reply:
WW3 is optional is S2S and Weather